tests/test_url_handler.py

```diff
@@ -0,0 +1,700 @@
+    url = factories.NominatedURLFactory(url_project=project)
+    nominator = url.url_nominator
+    returned = url_handler.create_url_dump(project)
+
+    assert returned == {
+        url.entity: {
+            'nominators': ['{0} - {1}'.format(
+                nominator.nominator_name,
+                nominator.nominator_institution
+            )],
+            'nomination_count': 1,
+            'nomination_score': int(url.value),
+            'attributes': {}
+        }
+    }
+
+
+def test_create_url_dump_url_surt():
+    project = factories.ProjectFactory()
+    surt = 'http://(com,example,www)'
+    domain_surt = 'http://(com,example,'
+    url = factories.SURTFactory(url_project=project, value=surt)
+    returned = url_handler.create_url_dump(project)
+
+    assert returned == {
+        url.entity: {
+            'nominators': [],
+            'nomination_count': 0,
+            'nomination_score': 0,
+            'attributes': {},
+            'surt': surt,
+            'domain_surt': domain_surt
+        }
+    }
+
+
+def test_create_url_dump_url_attribute():
+    project = factories.ProjectWithMetadataFactory(metadata2=None)
+    attribute = project.metadata.all()[0].name
+    value = models.Metadata_Values.objects.filter(metadata__name__iexact=attribute)[0].value
+    url = factories.URLFactory(url_project=project, attribute=attribute, value=value.key)
+    returned = url_handler.create_url_dump(project)
+
+    assert returned == {
+        url.entity: {
+            'nominators': [],
+            'nomination_count': 0,
+            'nomination_score': 0,
+            'attributes': {attribute: [value.value]},
+        }
+    }
+
+
+def test_create_url_dump_url_attribute_new_value_with_factory():
+    project = factories.ProjectWithMetadataFactory(metadata2=None)
+    attribute = project.metadata.all()[0].name
+    url = factories.URLFactory(url_project=project, attribute=attribute)
+    returned = url_handler.create_url_dump(project)
+
+    assert returned == {
+        url.entity: {
+            'nominators': [],
+            'nomination_count': 0,
+            'nomination_score': 0,
+            'attributes': {attribute: [url.value]},
+        }
+    }
+
+
+@pytest.mark.parametrize('surt_root, expected_letter', [
+    ('http://(com,example,www,)', False),
+    ('http://(com,a,', 'a')
+])
+def test_create_surt_dict(surt_root, expected_letter):
+    project = factories.ProjectFactory()
+    surts = [
+        surt_root + '/some/stuff',
+        surt_root + '/other/stuff',
+        surt_root + '/nothing',
+        surt_root
+    ]
+    urls = [factories.SURTFactory(url_project=project, value=surt) for surt in surts]
+    surt_dict = url_handler.create_surt_dict(project, surt_root)
+
+    assert len(surt_dict['url_list']) == len(surts)
+    for url in surt_dict['url_list']:
+        assert url in urls
+    assert surt_dict['letter'] == expected_letter
+
+
+def test_create_surt_dict_exception_caught():
```

Comment:
Also, I know you guys tend to avoid many docstrings and comments in tests, but if it is difficult to come up with a name that describes exactly what the test is, you could add a docstring for clarification.